test/aave-v2/TestSupplyVault.t.sol

```diff
@@ -261,14 +262,59 @@ contract TestSupplyVault is TestSetupVaults {
     function testPreviewDeposit() public {
         uint256 amount = 1e5 ether;
 
-        // This should use predicted update indexes.
-        uint256 previewBefore = daiSupplyVault.previewDeposit(amount);
+        // This should test that using the lens' predicted indexes is the correct amount to use.
+        uint256 preview1 = daiSupplyVault.previewDeposit(amount);
         vaultSupplier1.depositVault(daiSupplyVault, amount);
-        assertEq(daiSupplyVault.balanceOf(address(vaultSupplier1)), previewBefore, "before");
+        assertEq(daiSupplyVault.balanceOf(address(vaultSupplier1)), preview1, "before");
 
-        // The indexes should be the same as before this call.
-        uint256 previewAfter = daiSupplyVault.previewDeposit(amount);
+        // The mint interacts with Morpho which updates the indexes,
+        // so this should test that the lens predicted index does not differ from Morpho's actual index
+        uint256 preview2 = daiSupplyVault.previewDeposit(amount);
         vaultSupplier2.depositVault(daiSupplyVault, amount);
-        assertEq(daiSupplyVault.balanceOf(address(vaultSupplier2)), previewAfter, "after");
+        assertEq(daiSupplyVault.balanceOf(address(vaultSupplier2)), preview2, "after");
+    }
+
+    function testPreviewWithdraw() public {
+        uint256 amount = 1e5 ether;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount * 2);
+        vaultSupplier2.depositVault(daiSupplyVault, amount * 2);
+
+        uint256 balanceBefore1 = daiSupplyVault.balanceOf(address(vaultSupplier1));
+        uint256 balanceBefore2 = daiSupplyVault.balanceOf(address(vaultSupplier2));
+
+        vm.roll(block.number + 100);
```

Comment:
You need to warp for Aave as it relies on `block.timestamp` and not `block.number`